hive/replays/ppo_replay.py

```diff
@@ -0,0 +1,158 @@
+
+        Args:
+            capacity (int): Total number of observations that can be stored in the buffer
+            stack_size (int): The number of frames to stack to create an observation.
+            n_step (int): Horizon used to compute n-step return reward
+            gamma (float): Discounting factor used to compute n-step return reward
+            use_gae (bool): Whether to use generalised advantage estimates for calculating returns
+            gae_lambda (float): Discouting factor used to compute generalised advantage estimation
+            observation_shape: Shape of observations that will be stored in the buffer.
+            observation_dtype: Type of observations that will be stored in the buffer.
+                This can either be the type itself or string representation of the
+                type. The type can be either a native python type or a numpy type. If
+                a numpy type, a string of the form np.uint8 or numpy.uint8 is
+                acceptable.
+            action_shape: Shape of actions that will be stored in the buffer.
+            action_dtype: Type of actions that will be stored in the buffer. Format is
+                described in the description of observation_dtype.
+            action_shape: Shape of actions that will be stored in the buffer.
+            action_dtype: Type of actions that will be stored in the buffer. Format is
+                described in the description of observation_dtype.
+            reward_shape: Shape of rewards that will be stored in the buffer.
+            reward_dtype: Type of rewards that will be stored in the buffer. Format is
+                described in the description of observation_dtype.
+            extra_storage_types (dict): A dictionary describing extra items to store
+                in the buffer. The mapping should be from the name of the item to a
+                (type, shape) tuple.
+            num_players_sharing_buffer (int): Number of agents that share their
+                buffers. It is used for self-play.
+        """
+        if extra_storage_types is None:
+            extra_storage_types = dict()
+        extra_storage_types.update(
+            {
+                "values": (np.float32, ()),
+                "returns": (np.float32, ()),
+                "advantages": (np.float32, ()),
+                "logprob": (np.float32, ()),
+            }
+        )
+        super().__init__(
+            capacity + stack_size - 1,
+            stack_size,
+            n_step,
+            gamma,
+            observation_shape,
+            observation_dtype,
+            action_shape,
+            action_dtype,
+            reward_shape,
+            reward_dtype,
+            extra_storage_types,
+            num_players_sharing_buffer,
+        )
+        self._use_gae = use_gae
+        self._gae_lambda = gae_lambda
+
+    # Taken from https://github.com/vwxyzjn/ppo-implementation-details/blob/main/ppo_shared.py
```

Comment:
@dapatil211 Correct me if I'm wrong, but it would be better to get inspiration from others' code but write our own version. I know there aren't many ways to implement this function but we can probably at least reformat it.